maps/helpers.go

```diff
@@ -87,3 +91,91 @@ func (m StubMap) UpdateBoard(previousBoardState *rules.BoardState, settings rule
 	}
 	return nil
 }
+
+// drawRing draws a ring of hazard points offset from the outer edge of the board
+func drawRing(bw, bh, hOffset, vOffset int) ([]rules.Point, error) {
+	if bw < 1 {
+		return nil, errors.New("board width too small")
```

Comment:
Probably should be a `rules.RulesetError` here (and in the other checks too) so engine can detect an invalid configuration.